vpdq/cpp/regtest.py

```diff
@@ -73,90 +81,91 @@ def get_argparse() -> argparse.ArgumentParser:
-                        "-v",
-                        "-f",
-                        ffmpegPath,
-                        "-r",
-                        secondsPerHash,
-                        "-d",
-                        outputHashFolder,
-                        "-s",
-                        downsampleFrameDimension,
-                        "-i",
-                        inputVideoFolder + "/" + file,
-                    ]
-                )
-            else:
-                subprocess.call(
-                    [
-                        "./build/vpdq-hash-video",
-                        "-f",
-                        ffmpegPath,
-                        "-r",
-                        secondsPerHash,
-                        "-d",
-                        outputHashFolder,
-                        "-s",
-                        downsampleFrameDimension,
-                        "-i",
-                        inputVideoFolder + "/" + file,
-                    ]
-                )
-
-    cdir = os.getcwd()
-    pdir = os.path.dirname(cdir)
-    sample = pdir + ("/sample-hashes")
-    output = outputHashFolder
     distanceTolerance = str(args.matchDistanceTolerance)
     qualityTolerance = str(args.qualityTolerance)
-    for file in os.listdir(sample):
-        if file.endswith(".txt"):
-            print("\nMatching File " + file)
-            sampleFile = f"{sample}/{file}"
-            outputFile = f"{output}/{file}"
-            if verbose:
-                subprocess.call(
-                    [
-                        "./build/match-hashes-byline",
-                        "-v",
-                        sampleFile,
-                        outputFile,
-                        distanceTolerance,
-                        qualityTolerance,
-                    ]
-                )
-            else:
-                subprocess.call(
-                    [
-                        "./build/match-hashes-byline",
-                        sampleFile,
-                        outputFile,
-                        distanceTolerance,
-                        qualityTolerance,
-                    ]
-                )
+    verbose = args.verbose
+
+    outputHashFolder.mkdir(parents=True, exist_ok=True)
+
+    # TODO: Add more general options for other video extensions.
+    for fileStr in glob.iglob(f"{inputVideoFolder}/**/*.mp4", recursive=True):
+        file = Path(fileStr)
+
+        # Create output hash file or overwrite existing file
+        # This is hardcoded in cpp, and it
+        # does not create the file if it does not exist:
+        #
+        # vpdq-hash-video.cpp:
+        #
+        # // Strip containing directory:
+        # std::string b = basename(inputVideoFileName, "/");
+        # // Strip file extension:
+        # b = stripExtension(b, ".");
+        # outputHashFileName = outputDirectory + "/" + b + ".txt";
+
+        with open(outputHashFolder / f"{file.stem}.txt", "w"):
+            pass
```

Comment:
maybe blocking: I'm somewhat spooked by this script writing to files in an existing directory, this could be quite dangerous - what we can instead do is use pythons temporary directory functionality: https://docs.python.org/3/library/tempfile.html#tempfile.mkdtemp

NamedTemporaryFile allows you to specify `delete=False` to retain the data, but there's no equivalent - instead you can use /tmp/ `gettmpprefix()` https://docs.python.org/3/library/tempfile.html#tempfile.gettempprefix to get a random name that will get cleaned up. 

You can then output the created directory while the command is running as a hint of where to look at the output.

Reply:
Okay, I'll use a tempdir by default, but there's still an outputHashFolder parameter. If you want to use this instead of the tempdir, you would still have to write and potentially overwrite files in the directory.

I don't see how using a tempdir fixes this. Do you want me to use tempfiles if the user specifies outputHashFolder as well? But then if you want to inspect the files manually (I've done this multiple times, it's useful) you would have to not clean them up after a run. Then you have a bunch of tempfiles with random prefixes in your outputdir you have to delete manually. Yuck.

I would say check if a file exists before writing to it but then you can't run it multiple times without manually deleting all the files which is bad. Very annoying.

I think if we're using a tempdir it should be filled with regular files, not tempfiles. That way the file names retain their original structure as they should.

If the user specifies a folder, it should not be a tempdir, obviously.

And for either case there's no reason to clean up the files between runs because then you can't inspect them, so they should overwrite existing files (won't happen for tempdir) as they already do.

Reply:
I actually changed my mind and if a tempdir is used then just clean it up. This is probably the best option for 99% of tests where you just want to see it pass.

You can create TemporaryDirectory's with the context manager that aren't deleted starting in Python 3.12, but not now. Not super eager to use tempfile.mkdtemp directly.